sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -0,0 +1,1422 @@
+        private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly string _subscriptionKey;
+        private readonly string _apiVersion;
+
+        /// <summary>
+        /// Protected constructor to allow mocking.
+        /// </summary>
+        protected TextAnalyticsClient()
+        {
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="credential"></param>
+        public TextAnalyticsClient(Uri endpoint, TokenCredential credential)
+            : this(endpoint, credential, new TextAnalyticsClientOptions())
+        {
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="credential"></param>
+        /// <param name="options"></param>
+        public TextAnalyticsClient(Uri endpoint, TokenCredential credential, TextAnalyticsClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNull(credential, nameof(credential));
+            Argument.AssertNotNull(options, nameof(options));
+
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="subscriptionKey"></param>
+        public TextAnalyticsClient(Uri endpoint, string subscriptionKey)
+            : this(endpoint, subscriptionKey, new TextAnalyticsClientOptions())
+        {
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="subscriptionKey"></param>
+        /// <param name="options"></param>
+        public TextAnalyticsClient(Uri endpoint, string subscriptionKey, TextAnalyticsClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNullOrEmpty(subscriptionKey, nameof(subscriptionKey));
+            Argument.AssertNotNull(options, nameof(options));
+
+            _baseUri = endpoint;
+            _subscriptionKey = subscriptionKey;
+            _apiVersion = options.GetVersionString();
+            _pipeline = HttpPipelineBuilder.Build(options);
+            _clientDiagnostics = new ClientDiagnostics(options);
+        }
+
+        #region Detect Language
+
+        // Note that this is a simple overload that takes a single input and returns a single detected language.
+        // More advanced overloads are available that return the full list of detected languages.
+        /// <summary>
+        /// </summary>
+        /// <param name="inputText"></param>
+        /// <param name="countryHint"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual async Task<Response<DetectedLanguage>> DetectLanguageAsync(string inputText, string countryHint = "us", CancellationToken cancellationToken = default)
```

Comment:
1. Really default to "us"?
2. Is there any "database" of valid values? Maybe we should have an extensible enum or something here?

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/8780